corehq/apps/sms/views.py

```diff
@@ -1639,46 +1638,46 @@ def edit_sms_languages(request, domain):
     """
     Accepts same post body as corehq.apps.app_manager.views.edit_app_langs
     """
-    with StandaloneTranslationDoc.get_locked_obj(domain, "sms",
-        create=True) as tdoc:
+    with transaction.atomic():
+        sms_translations = get_or_create_sms_translations(domain)
         try:
             from corehq.apps.app_manager.views.utils import validate_langs
-            langs, rename = validate_langs(request, tdoc.langs)
+            langs, rename = validate_langs(request, sms_translations.langs)
         except AssertionError:
             return HttpResponse(status=400)
 
         for old, new in rename.items():
             if old != new:
-                tdoc.translations[new] = tdoc.translations[old]
-                del tdoc.translations[old]
+                sms_translations.translations[new] = sms_translations.translations[old]
+                del sms_translations.translations[old]
 
         for lang in langs:
-            if lang not in tdoc.translations:
-                tdoc.translations[lang] = {}
+            if lang not in sms_translations.translations:
+                sms_translations.translations[lang] = {}
```

Comment:
`nit`: Could remove the `if` line by  doing `sms_translations.translations[lang] = sms_translations.translations.get(lang, {})`

Reply:
leaving this one alone, the suggested code is shorter but I think counterintuitive to read